### DISCUSSION TITLE
Unable to Upgrade to 87+

### DISCUSSION THREAD

**Discussion by ben-page** [cefsharp/CefSharp#3430] [Q&A]
I'm unable to upgrade my .NET Framework 4.8.2 Windows Forms project from CefSharp 86 to 87 or 88. Compiling as AnyCPU causes the following error. 

`System.BadImageFormatException: 'Could not load file or assembly 'CefSharp.Core.Runtime, Version=88.2.90.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138' or one of its dependencies. An attempt was made to load a program with an incorrect format.'`

It works when I switch to 64bit. One possible quirk. We are using SDK style projects with .NET Framework 4.8.2 and Windows Forms.  But everything works perfectly with 86.

Can anyone provide assistance?

**Comment by amaitland** [comment]
As a short term workaround add the following to your project file

```xml
<PropertyGroup>
  <CefSharpPlatformTargetOverride>AnyCPU</CefSharpPlatformTargetOverride>
</PropertyGroup>
```

The condition at https://github.com/cefsharp/CefSharp/blob/cefsharp/89/NuGet/CefSharp.Common.targets#L110 needs a little tweak. This is a workaround for https://github.com/dotnet/sdk/issues/1560

This should be fixed in `89`.

**Reply by amaitland** [reply]
After doing a little more digging I don't think hacking around this is the correct approach,  there are too many variances between the different visual studio versions and .net SDK installs.

I'll likely just generate an error in the event the PlatformTarget differs for SDK style projects.

**Reply by TonyValenti** [reply]
In v89, will I be able to use a packagereference to include both the x86 and x64 resources?

**Reply by amaitland** [reply]
> will I be able to use a packagereference to include both the x86 and x64 resources?

`PackageReference` is not specific to the newer `SDK Style` project format, it is usable with older `Non SDK Style` projects as well. 

So I'm going to assume you are referring to a newer `SDK Style` project targeting `AnyCPU`. You will need to specify `PlatformTarget` in your project file as per https://github.com/dotnet/sdk/issues/1560#issuecomment-327043099.

Using `CefSharpPlatformTargetOverride` will also still work (as it does with version `88`)

```xml
<PlatformTarget>AnyCPU</PlatformTarget>
```

**Reply by amaitland** [reply]
The [89.0.140-pre](https://github.com/cefsharp/CefSharp/issues/3446#issuecomment-806380636) release is on `Nuget.org` which should generate an `MSBuild` error for `SDK Style` projects that don't correctly specify `PlatformTarget` the error message is at https://github.com/cefsharp/CefSharp/blob/cefsharp/89/NuGet/CefSharp.Common.targets#L380

Suggestions/improvements on wording to make it clearer are welcome.

Let me know how it goes.
